processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -54,25 +56,46 @@ public DoubleMaxAggregatorFactory(String name, String fieldName)
   @Override
   public Aggregator factorize(ColumnSelectorFactory metricFactory)
   {
-    return new DoubleMaxAggregator(getDoubleColumnSelector(metricFactory, Double.NEGATIVE_INFINITY));
+    BaseDoubleColumnValueSelector doubleColumnSelector = getDoubleColumnSelector(
+        metricFactory,
+        Double.NEGATIVE_INFINITY
+    );
+    return NullHandlingHelper.getNullableAggregator(
```

Comment:
As it it done in *every* `AggregatorFactory`, and it's impossible to change agg factories in proprietary extensions, and authors of new AggregatorFactories will forget to do this all the time, such null handling must be added in some centralized way.

Reply:
refactored to an abstract class NullAggregatorFactory. Please check again.